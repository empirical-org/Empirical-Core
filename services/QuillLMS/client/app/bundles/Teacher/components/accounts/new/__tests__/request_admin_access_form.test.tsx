import { mount } from 'enzyme';
import * as React from 'react';

import RequestAdminAccessForm from '../request_admin_access_form';

describe('RequestAdminAccessForm component', () => {

  it('should render', () => {
    const component = mount(<RequestAdminAccessForm admins={[{ name: 'Joy Harjo', email: 'joyharjo@quill.org', id: 1 }]} schoolName="School Name" />);
    expect(component).toMatchSnapshot();
  });

});
