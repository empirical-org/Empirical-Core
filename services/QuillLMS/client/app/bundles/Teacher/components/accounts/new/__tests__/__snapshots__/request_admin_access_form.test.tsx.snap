// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestAdminAccessForm component should render 1`] = `
<RequestAdminAccessForm
  admins={
    Array [
      Object {
        "email": "joyharjo@quill.org",
        "id": 1,
        "name": "Joy Harjo",
      },
    ]
  }
  schoolName="School Name"
>
  <img
    alt=""
    src="undefined/images/onboarding/school_verification.svg"
  />
  <h1>
    <span>
      Please request permission to become an admin of:
    </span>
    <span
      className="school-name"
    >
      School Name
    </span>
  </h1>
  <p
    className="sub-header"
  >
    Which admin(s) do you want to send the request to?
  </p>
  <section
    className="user-account-section admin-access-request-section"
  >
    <AdminTable
      schoolAdmins={
        Array [
          Object {
            "email": "joyharjo@quill.org",
            "id": 1,
            "name": "Joy Harjo",
          },
        ]
      }
      selectedAdminIds={Array []}
      setSelectedAdminIds={[Function]}
    >
      <div
        className="admin-selection-table"
      >
        <div
          className="admin-selection-row top-row"
        >
          <button
            aria-label="Select all admins"
            className="focus-on-light quill-checkbox unselected"
            onClick={[Function]}
            type="button"
          />
          <div>
            Select all
          </div>
        </div>
        <div
          className="admin-selection-row"
          key="1"
        >
          <button
            aria-label="Select Joy Harjo"
            className="focus-on-light quill-checkbox unselected"
            onClick={[Function]}
            type="button"
          />
          <div>
            <span
              className="name"
            >
              Joy Harjo
            </span>
            <span
              className="email"
            >
              joyharjo@quill.org
            </span>
          </div>
        </div>
      </div>
    </AdminTable>
    <Input
      autoComplete="on"
      className="reason"
      handleChange={[Function]}
      label="Why do you want to be an admin?"
      value=""
    >
      <div
        className="input-container  inactive  reason "
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={-1}
      >
        <label>
          Why do you want to be an admin?
        </label>
        <input
          autoComplete="on"
          maxLength={10000}
          onChange={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
    </Input>
  </section>
  <div
    className="button-wrapper"
  >
    <button
      className="quill-button primary contained medium focus-on-light disabled"
      disabled={true}
      type="button"
    >
      Submit
    </button>
  </div>
  <SkipForNow>
    <a
      className="skip-for-now"
      href="/finish_sign_up"
    >
      Skip for now
    </a>
  </SkipForNow>
</RequestAdminAccessForm>
`;
